### ISSUE TITLE
missing classmethod decorator on from_pretrained methods

### ISSUE THREAD

**Issue by FlorianMF** [PyTorchLightning/lightning-bolts#597]
## 🐛 Bug

The method from_pretrained in at least AE (basic_ae_module.py) and VAE (basic_vae_module.py) should be a classmethod to instantiate an object using this method. 
Momentarily calling this method returns TypeError: from_pretrained() missing 1 required positional argument: 'checkpoint_name'.

### Environment

 - Bolts: 0.3.0

**Comment by FlorianMF** [comment]
In the fixed issue https://github.com/PyTorchLightning/lightning-bolts/issues/224 the model has first to be instantiated and than the method called.
The docstrings though state the following :
```
    Example::

        # not pretrained
        ae = AE()

        # pretrained on cifar10
        ae = AE.from_pretrained('cifar10-resnet18')
```

**Comment by akihironitta** [comment]
@FlorianMF Thank you for reporting the issue. We recently fixed the docs in #557, so could you have a look at the latest docs?
https://pytorch-lightning-bolts.readthedocs.io/en/latest/autoencoders.html

related: https://github.com/PyTorchLightning/lightning-bolts/issues/200

Feel free to reopen this issue if there's still any problem.
